README.rst

```diff
@@ -110,29 +110,23 @@ Documentation
 
 Installation
 ------------
-::
 
-    pip install smart_open  // Install with no cloud dependencies
-    pip install smart_open[s3] // Install S3 deps
-    pip install smart_open[gcp] // Install GCP deps
-    pip install smart_open[azure] // Install Azure deps
-    pip install smart_open[all] // Installs all cloud dependencies
+``smart_open`` supports a wide range of storage solutions, including AWS S3, Google Cloud and Azure.
+Each individul solution has its own dependencies.
+By default, ``smart_open`` does not install any dependencies, in order to keep the installation size small.
+You can install these dependencies explicitly using::
 
-Or, if you prefer to install from the `source tar.gz <http://pypi.python.org/pypi/smart_open>`_::
+    pip install smart_open[azure] # Install Azure deps
+    pip install smart_open[gcp] # Install GCP deps
+    pip install smart_open[s3] # Install S3 deps
 
-    python setup.py test  # run unit tests
-    python setup.py install
+Or, if you don't mind installing a large number of third party libraries, you can install all dependencies using::
 
-To run the unit tests (optional), you'll also need to install some other dependencies: see setup.py or run `pip install .[test]`.
-The tests are also run automatically with `Travis CI <https://travis-ci.org/RaRe-Technologies/smart_open>`_ on every commit push & pull request.
-
-If you're upgrading from ``smart_open`` versions 1.8.0 and below, please check out the `Migration Guide <MIGRATING_FROM_OLDER_VERSIONS.rst>`_.
+    pip install smart_open[all]
 
-Version ``3.0`` will introduce a backwards incompatible installation method with regards to the cloud dependencies.
-If you want to maintain backwards compatibility (installing all dependencies) install this package via ``smart_open[all]`` now
-and once the change is made you should not have any issues. If all you care about is AWS dependencies for example you can install via ``smart_open[s3]`` and
-once the dependency change is made you will simply drop the unwanted dependencies. You can read more about the motivations `here <https://github.com/RaRe-Technologies/smart_open/issues/443>`_
+Be warned that this option increases the installation size significantly, e.g. over 100MB.
```

Comment:
Oh wow. Is this true? Which library / dependency is the main culprit?

Reply:
botocore is around 50MB. Other cloud dependencies like boto3, azure and google are smaller (several MB) but it adds up in the end.